src/components/Pagination/PaginationRow.js

```diff
@@ -0,0 +1,166 @@
+      'card-view-pf-pagination': cardViewPagination,
+      'table-view-pf-pagination': tableViewPagination,
+      clearfix: true
+    },
+    className
+  );
+  return (
+    <Form className={classes}>
+      <FormGroup>
+        <DropdownButton
+          title={perPage}
+          dropup
+          componentClass={PaginationRowButtonGroup}
+          onSelect={onPerPageSelect}
+          id={dropdownButtonId}
+        >
+          {perPageOptions &&
+            perPageOptions.length &&
+            perPageOptions.map((option, i) => {
+              return (
+                <MenuItem eventKey={option} active={option === perPage} key={i}>
+                  {option}
+                </MenuItem>
+              );
+            })}
+        </DropdownButton>
+        <span> {messages.perPage} </span>
+      </FormGroup>
+      <FormGroup>
+        <PaginationRowItems
+          itemCount={itemCount}
+          itemsStart={itemsStart}
+          itemsEnd={itemsEnd}
+          messagesOf={messages.of}
+        />
+
+        <PaginationRowBack
+          page={page}
+          messagesFirstPage={messages.firstPage}
+          messagesPreviousPage={messages.previousPage}
+          onFirstPage={onFirstPage}
+          onPreviousPage={onPreviousPage}
+        />
+
+        <ControlLabel className="sr-only">Current Page</ControlLabel>
+        <FormControl
+          className="pagination-pf-page"
+          type="text"
+          value={pagination.page}
+          onChange={onPageInput}
+        />
+
+        <PaginationRowAmountOfPages
+          messagesOf={messages.of}
+          amountOfPages={amountOfPages}
+        />
+
+        <PaginationRowForward
+          page={page}
+          amountOfPages={amountOfPages}
+          messagesNextPage={messages.nextPage}
+          messagesLastPage={messages.lastPage}
+          onNextPage={onNextPage}
+          onLastPage={onLastPage}
+        />
+      </FormGroup>
+    </Form>
+  );
+};
+PaginationRow.propTypes = {
+  /** Additional css classes */
+  className: PropTypes.string,
+  /** user pagination settings */
+  pagination: PropTypes.object,
+  /** calculated amount of pages */
+  amountOfPages: PropTypes.number,
+  /** calculated number of rows */
+  itemCount: PropTypes.number,
+  /** calculated items start */
+  itemsStart: PropTypes.number,
+  /** calculated items end */
+  itemsEnd: PropTypes.number,
+  /** message text inputs for i18n */
+  messages: PropTypes.shape({
+    firstPage: PropTypes.string,
+    previousPage: PropTypes.string,
+    nextPage: PropTypes.string,
+    lastPage: PropTypes.string,
+    perPage: PropTypes.string,
+    of: PropTypes.string
+  }),
+  /** content view pagination row */
+  contentViewPagination: PropTypes.bool,
+  /** table view pagination row */
+  tableViewPagination: PropTypes.bool,
+  /** list view pagination row */
+  listViewPagination: PropTypes.bool,
+  /** card view pagination row */
+  cardViewPagination: PropTypes.bool,
```

Comment:
It would be better to use a viewType enum 'list', 'card', 'table' since these are mutually exclusive.

Reply:
gotcha...will do!

Reply:
added `PAGINATION_VIEW_TYPES`.